### ISSUE TITLE
90

### ISSUE THREAD

**Issue by vaskonov** [deep-nlp-spring-2020/Task-2-Tatiana-Golubeva#1]
**Assignment 2.1**
15/15

**Assignment 2.2**
25/35
2.1 5/5
2.2 1/1
2.3 1/1
2.4 3/3
2.5 3/3
2.6-7 2/6 the probability values are weird. I am not sure what you do exactly, but it seems that you propagate hidden states from the last step for the first batch to the first step for the second, which is very strange.
2.8 5/5 Your do not actually need to check the predictions of `begin` and `pad` symbols, the trained model is clever enough not to predict them.
2.9 1/1
2.10 1/5 you are still sampling letters for all words if the batch even when the 'end' symbol is generated. Try to maintain the set of active indexes and slice your states and outputs  ob every step of your first loop.
2.11 3/5 For me it looks quite strange that val_loss stabilizes with much lower value than train_loss does. Probably, your metrics are not correct. Try to check it by choosing the same train and validation sets.

**Assignment 2.3**

- Creating the Iterator (2 points)
2 from 2

- Define the RNN-based text classification model (10 points)
Here there is no need to strictly follow the schema
very nice trick with 
`hidden = torch.cat((hidden[-2,:,:], hidden[-1,:,:]), dim = 1)`

- The training loop (3 points)
3 from 3
Here make sure that the model is not over-fitted

- Calculate performance of the trained model (5 points)
5 from 5
nicely done

- Experiments (10 points)
10 points 10

**Assignment 2.4**

- Creating the Iterator (2 points)
  2 points from 2

- Define CNN-based text classification model (8 points)
  8 points from 8

- The training loop (3 points)
  3 points from 3
  Again make sure that the model is not overfitted here

-  Calculate performance of the trained model (2 points)
   2 points from 2

- Experiments (5 points)
  5 points from 5
